### PR TITLE
deprecate string methods of character predicates, e.g. `isnumber("")`

### LINKED ISSUE #14156: isnumber("")  return true
```
julia> isnumber("")
true

julia> versioninfo()
Julia Version 0.4.1
Commit cbe1bee* (2015-11-08 10:33 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) CPU E5-2630 v2 @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

```

### PR DESCRIPTION
**Pull request by JeffBezanson**
fixes #14156

### CODE REVIEW

File: test/unicode/utf8proc.jl [hunk 1]
```diff
@@ -187,34 +187,33 @@ let
...
-@test iscntrl("ΣβΣβ")==false
-@test ispunct("ΣβΣβ")==false
-
-@test isnumber("23435")==true
-@test isdigit("23435")==true
-@test isalnum("23435")==true
-@test isalpha("23435")==false
-@test iscntrl( string(Char(0x0080))) == true
-@test ispunct( "‡؟჻") ==true
+@test all(isspace,"  \t   \n   \r  ")==true
```
Comment (nalimilan) [inline test/unicode/utf8proc.jl]: BTW, maybe change `==` into `===` or remove them altogether?

File: base/strings/basic.jl [hunk 2]
```diff
@@ -280,20 +280,19 @@ isascii(s::AbstractString) = all(isascii, s)
...
-Tests whether a character is a valid hexadecimal digit, or whether this is true for all elements of a string.
+Tests whether a character is a valid hexadecimal digit.
 
 ```jldoctest
-julia> isxdigit("abc")
+julia> isxdigit('a')
 true
 
-julia> isxdigit("0x9")
+julia> isxdigit('z')
```
Comment (nalimilan) [inline base/strings/basic.jl]: The `0` and `x` cases where interesting to show that the prefix wasn't considered as a valid hexadecimal digit. Say this in the docstring?

File: base/deprecated.jl [hunk 3]
```diff
@@ -1849,4 +1849,10 @@ end)
 
 @deprecate FloatRange{T}(start::T, step, len, den) Base.floatrange(T, start, step, len, den)
 
+for name in ("alnum", "alpha", "cntrl", "digit", "number", "graph",
+             "lower", "print", "punct", "space", "upper", "isxdigit")
```
Comment (StefanKarpinski) [inline base/deprecated.jl]: I think you want just `"xdigit"` for the last one.

**Review by StefanKarpinski — CHANGES_REQUESTED** [review]

**Review by ararslan — APPROVED** [review]
👍

**Pull request merged**